### PR TITLE
Add CI to build notebooks

### LINKED ISSUE #2: Continuous Integration
This repository could use continuous integration to ensure that the examples continue to work over time

I'm not familiar with tools to test notebooks, but other projects do this, so it must be doable.

### PR DESCRIPTION
**Pull request by mholtzscher**
Travis configuration and script to execute all notebooks in repo.

Fixes #2

### CODE REVIEW

**Comment by mholtzscher** [comment]
Example Build: https://travis-ci.org/mholtzscher/dask-examples

File: tests/nbrun.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+def run_notebook(in_filepath):
+    if not in_filepath.is_file():
+        raise IOError('File "%s" not found.' % in_filepath)
+    in_filepath = in_filepath.resolve()
+
+    run_path = str(in_filepath.parent)
+
+    nb = nbformat.read(in_filepath.open(), as_version=4)
+    ep = ExecutePreprocessor(
+        timeout=3600, kernel_name='python%d' % sys.version_info[0])
```
Comment (mrocklin) [inline tests/nbrun.py]: The timeout here is in seconds?  If so then I suggest that we reduce this a bit to something like 300 (five minutes)

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+pandas
+scikit-learn
+matplotlib
+h5py
+graphviz
+nbconvert
+nbformat
+pathlib
+jupyter_client
+ipykernel
```
Comment (mrocklin) [inline requirements.txt]: We currently keep an environment.yml file that we use with the Binder deployment.  I wonder if we can reuse that here instead?  I worry a bit about having to keep the two in sync.  

I tend to use miniconda for testing.  I'm not sure if we want to do that here, but if so I recommend stealing the following lines: https://github.com/mrocklin/streamz/blob/c180551ff5a7eb7aa30bf56242554144f726d45a/.travis.yml#L17-L22

and then following up with 

```
conda env create -f environment.yaml -n test
source activate test
```

File: tests/nbrun.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        msg = 'Error executing the notebook "%s".\n\n' % in_filepath
+        print(msg)
+        raise
+
+
+print("Running Notebooks...")
+
+for filename in glob.glob('../**/*.ipynb', recursive=True):
+    print(filename)
+    run_notebook(Path(filename))
```
Comment (mrocklin) [inline tests/nbrun.py]: I'm curious, was this code inspired by another project's solution or is it from scratch?

File: tests/nbrun.py (hunk 1 above)
Comment (mholtzscher) [reply tests/nbrun.py]: Yes, it is in seconds. I meant to change it but forgot. I agree with 300.

File: requirements.txt (hunk 2 above)
Comment (mholtzscher) [reply requirements.txt]: I will investigate using miniconda instead

File: tests/nbrun.py (hunk 3 above)
Comment (mholtzscher) [reply tests/nbrun.py]: I looked at a couple other repos and combined what I liked from each

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+language: python
+
+python:
+    - "3.5"
+    - "3.6"
```
Comment (mholtzscher) [inline .travis.yml]: @mrocklin  what python versions do you want me to target?

Comment (mrocklin) [reply .travis.yml]: Perhaps controversially, I suggest that we only do 3.6 for the examples

**Comment by TomAugspurger** [comment]
Thanks @mholtzscher.

Would you mind (temporarily) adding an error in a notebook and ensure that the failure bubbles all the way up to fail the build? Everything looks like it should, but just want to verify.

**Comment by mholtzscher** [comment]
@TomAugspurger Here you go! https://travis-ci.org/mholtzscher/dask-examples/builds/362830883

**Comment by TomAugspurger** [comment]
Great, thanks! Happy to see the failure :)

**Comment by mrocklin** [comment]
This looks good to me.  I've taken the liberty of making a small change and moving the extra dependencies to the travis.yml file rather than the environment.yml file, which is also shared with the binder deployment, which we'd like to keep lean if possible.

**Comment by mrocklin** [comment]
Merged.  Thanks @mholtzscher !

**Pull request merged**